orderer/common/cluster/util_test.go

```diff
@@ -1112,3 +1116,280 @@ func TestChainParticipant(t *testing.T) {
 		})
 	}
 }
+
+func TestCompareCertPublicKeysWithSameKeyDifferentBytes(t *testing.T) {
+	t.Parallel()
+
+	// Create a CA for generating certificates
+	ca, err := tlsgen.NewCA()
+	require.NoError(t, err)
+
+	// Generate first certificate
+	firstKeyPair, err := ca.NewClientCertKeyPair()
+	require.NoError(t, err)
+	firstCert, err := fabriccrypto.SanitizeX509Cert(firstKeyPair.Cert)
+	require.NoError(t, err)
+
+	// Generate second certificate using the same private key (same public key)
+	// but different certificate attributes (different serial number, timestamps, etc.)
+	secondCert, err := generateCertWithSameKeyForTesting(firstKeyPair.Signer, ca)
+	require.NoError(t, err)
+
+	// Verify that bytes.Equal fails - certificates should have different bytes
+	require.False(t, bytes.Equal(firstCert, secondCert),
+		"Certificates should have different bytes due to different serial numbers/timestamps")
+
+	// Test the actual CompareCertPublicKeys function from cluster package
+	equal, err := cluster.CompareCertPublicKeys(firstCert, secondCert)
+	require.NoError(t, err, "CompareCertPublicKeys should not return an error for valid certificates")
+	require.True(t, equal, "CompareCertPublicKeys should return true for certificates with same public key")
+
+	// Also verify manually that public keys are the same (for additional validation)
+	block1, _ := pem.Decode(firstCert)
+	require.NotNil(t, block1, "First certificate should be valid PEM")
+	cert1, err := x509.ParseCertificate(block1.Bytes)
+	require.NoError(t, err)
+
+	block2, _ := pem.Decode(secondCert)
+	require.NotNil(t, block2, "Second certificate should be valid PEM")
+	cert2, err := x509.ParseCertificate(block2.Bytes)
+	require.NoError(t, err)
+
+	// Extract public keys
+	pubKey1, err := x509.MarshalPKIXPublicKey(cert1.PublicKey)
+	require.NoError(t, err)
+	pubKey2, err := x509.MarshalPKIXPublicKey(cert2.PublicKey)
+	require.NoError(t, err)
+
+	// Verify that public keys are the same even though certificate bytes differ
+	require.True(t, bytes.Equal(pubKey1, pubKey2),
+		"Public keys should be identical even though certificate bytes differ")
+
+	// This test demonstrates the scenario described in the PR comment:
+	// - bytes.Equal(firstCert, secondCert) returns false (different certificate bytes)
+	// - But the public keys are the same, so compareCertPublicKeys would return true
+	// - This ensures that certificate renewal/reissuance with same key pair works correctly
```

Comment:
Renewal means extending the notAfter right? this ignores every aspect of the cert including key usage, name, etc.

Reply:
That's true, I think ideally he would just copy the certificate itself and just change the notAfter/notBefore, but I thought the gist of the test is clear so I let it go.